src/test/java/teammates/test/driver/PriorityInterceptor.java

```diff
@@ -34,86 +34,82 @@
-                return mi.getMethod().getMethod().getDeclaringClass().getName();
-            }
-
-            private int packagePriorityOffset(String packageName) {
-                int index = packageOrder.indexOf(packageName);
-
-                if (index == -1) {
-                    return 0;
-                }
-                return -index;
-            }
-
-            @Override
-            public int compare(IMethodInstance m1, IMethodInstance m2) {
-                int val = 0;
-
-                //Compare by package name
-                String p1 = getPackageName(m1);
-                String p2 = getPackageName(m2);
-                val = p1.compareTo(p2);
-                val -= packagePriorityOffset(p1);
-                val += packagePriorityOffset(p2);
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by class priority
-                val = getClassPriority(m1) - getClassPriority(m2);
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by class name
-                val = getClassName(m1).compareTo(getClassName(m2));
-                if (val != 0) {
-                    return val;
-                }
-
-                //Compare by method priority
-                val = getMethodPriority(m1) - getMethodPriority(m2);
-                if (val != 0) {
-                    return val;
-                }
-
-                return 0;
-            }
+                                           ITestContext context) {
+
+        //Compare by package name
+        Comparator<IMethodInstance> compareByPackage = (IMethodInstance m1, IMethodInstance m2) -> {
+            int val = 0;
 
+            String p1 = getPackageName(m1);
+            String p2 = getPackageName(m2);
+            val = p1.compareTo(p2);
+            val -= packagePriorityOffset(p1);
+            val += packagePriorityOffset(p2);
+
+            return val;
         };
+
+        //Compare by class priority
+        Comparator<IMethodInstance> compareByClassPriority = Comparator.comparing((IMethodInstance m) ->
+                getClassPriority(m));
+
+        //Compare by class name
+        Comparator<IMethodInstance> compareByClassName = Comparator.comparing((IMethodInstance m) ->
+                getClassName(m));
+
+        //Compare by class name
+        Comparator<IMethodInstance> compareByMethodPriority = Comparator.comparing((IMethodInstance m) ->
+                getMethodPriority(m));
+
+        //Overall Comparator sorts in the order of package name -> class priority -> class name -> method priority
+        Comparator<IMethodInstance> comparator = compareByPackage.thenComparing(compareByClassPriority)
+                .thenComparing(compareByClassName)
+                .thenComparing(compareByMethodPriority);
+
         IMethodInstance[] array = methods.toArray(new IMethodInstance[methods
```

Comment:
Not your fault, but the next three lines is basically converting to array -> sorting -> converting back to list. This looks redundant as we can just sort the list itself. Let's fix it.